ci/windows/build_common.psm1

```diff
@@ -0,0 +1,68 @@
+
+param ($CXX_STANDARD,
+       $GPU_ARCHS)
+
+$script:HOST_COMPILER = "cl"
+$script:PARALLEL_LEVEL = (Get-WmiObject -class Win32_processor).NumberOfLogicalProcessors
+
+If($null -eq $DEVCONTAINER_NAME) {
+    $script:BUILD_DIR="../build/local"
+} else {
+    $script:BUILD_DIR="../build/${DEVCONTAINER_NAME}"
+}
+
+If(!(test-path -PathType container "../build")) {
+    New-Item -ItemType Directory -Path "../build"
+}
+
+# The most recent build will always be symlinked to cccl/build/latest
+New-Item -ItemType Directory    -Path "$BUILD_DIR" -Force
+New-Item -ItemType SymbolicLink -Path "../build/latest" -Target "$BUILD_DIR" -Force
+
+$script:COMMON_CMAKE_OPTIONS= @(
+    "-S .."
+    "-B $BUILD_DIR"
+    "-G ""Ninja"""
+    "-DCMAKE_BUILD_TYPE=Release"
+    "-DCMAKE_CXX_STANDARD=${CXX_STANDARD}"
+    "-DCMAKE_CUDA_STANDARD=${CXX_STANDARD}"
+    "-DCMAKE_CXX_COMPILER=${HOST_COMPILER}"
+    "-DCMAKE_CUDA_HOST_COMPILER=${HOST_COMPILER}"
+    "-DCMAKE_CUDA_ARCHITECTURES=${GPU_ARCHS}"
+    "-DCMAKE_CUDA_COMPILER=nvcc"
+    "-DCMAKE_CUDA_COMPILER_FORCED=ON"
+    "-DCMAKE_EXPORT_COMPILE_COMMANDS=ON"
+)
```

Comment:
Note to future self, we can likely eliminate this redundancy between the linux and windows scripts by using cmake presets for things like this: https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html

Reply:
Filed https://github.com/NVIDIA/cccl/issues/252